packages/core/content-manager/server/controllers/relations.js

```diff
@@ -55,6 +55,14 @@ module.exports = {
 
     await validateFindAvailable(ctx.request.query);
 
+    const permissionChecker = getService('permission-checker').create({ userAbility, model });
+
+    if (permissionChecker.cannot.read()) {
+      return ctx.forbidden();
+    }
+
+    const permissionQuery = await permissionChecker.sanitizedQuery.read(ctx.request.query);
+
```

Comment:
Uppon further inspection I realized our desired behaviour when the user doesnt have permissions to see a relation is not to completely hide it, but rather hide the main field.

At the moment this was working for normal permissions but not for conditional ones. (You can see the behaviour in the `sanitizeMainField` function above.

Let me give it a thought this week! And see if we can come with another solution, or we commit to your approach (which will need product aproval)

Thank you for the contribution Jim, glad to see you contributing 🚀 
Give me another day and we will come with an answer 🙏 

cc @innerdvations

Reply:
agree, I am waiting for a product team response ( lots of holidays during december 🙏 ).